google-cloud-firestore/src/test/java/com/google/cloud/firestore/TransactionTest.java

```diff
@@ -213,6 +298,34 @@ public String updateCallback(Transaction transaction) {
     assertEquals(1, requests.size());
   }
 
+  @Test
+  public void noRollbackOnBeginFailureAsync() throws Exception {
```

Comment:
You didn't change this code path, so I think we can remove this test.

Reply:
I duplicate all tests in TransactionTest.java that somehow play with runTransaction and implemented them with runAsyncTransaction. Old sync/blocking methods as it is now implemented also are testing async implementation. I can delete all of *async tests. Can I delete all *async  tests?

Reply:
Almost all *async tests were deleted except those with other mocked setup